gci/src/main/java/com/danielfireman/gci/Sampler.java

```diff
@@ -0,0 +1,49 @@
+package com.danielfireman.gci;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class Sampler {
+	// Default sample rate should be fairly small, so big requests get checked up quickly.
+	private final int DEFAULT_SAMPLE_RATE = 10; // TODO(David) Update this value, if needed
+	// Max sample rate can not be very big because of peaks.
+	// The algorithm is fairly conservative, but we never know.
+	private final int MAX_SAMPLE_RATE = 30; // TODO(David) Update this value, if needed
+	
+	private int next;
+	private int[] pastSampleRates;
```

Comment:
For consistency, please pick one: sampleRate or sampleSize. It is quite bad to have pastSampleRates and currentSampleSize :)

Reply:
You mean pick one to replace currentSampleSize, right?

Reply:
Either call it pastSampleSize or rename currentSampleRate to currentSizeSize.

Reply:
Done